test/e2e/apimachinery/garbage_collector.go

```diff
@@ -576,18 +576,34 @@ var _ = SIGDescribe("Garbage collector", func() {
 		}
 		// wait for deployment to create some rs
 		ginkgo.By("Wait for the Deployment to create new ReplicaSet")
+		var replicaset appsv1.ReplicaSet
 		err = wait.PollImmediate(500*time.Millisecond, 1*time.Minute, func() (bool, error) {
 			rsList, err := rsClient.List(context.TODO(), metav1.ListOptions{})
 			if err != nil {
 				return false, fmt.Errorf("failed to list rs: %v", err)
 			}
-			return len(rsList.Items) > 0, nil
+			if len(rsList.Items) > 0 {
+				replicaset = rsList.Items[0]
+				return true, nil
+			}
+			return false, nil
 
 		})
 		if err != nil {
 			framework.Failf("Failed to wait for the Deployment to create some ReplicaSet: %v", err)
 		}
 
+		desiredGeneration := replicaset.Generation
+		if err := wait.PollImmediate(100*time.Millisecond, 60*time.Second, func() (bool, error) {
```

Comment:
100ms seems aggressively short to me, but I see there are other tests in apimachinery that use this interval

Reply:
I think I just copy &pasted it from somewhere around here 😀